### ISSUE TITLE
Global memcached groups names

### ISSUE THREAD

**Issue by rebeccahum** [Automattic/VIP-Coding-Standards#260]
## Bug Description
We should not allow use of global memcached cache group names, as they are being used on a global scale and things go wrong (such as user login) when they are overwritten: https://github.com/Automattic/wp-memcached

## Minimal Code Snippet

```php
wp_cache_set( 'blahblah', 'blah', 'users', 1 * MINUTE_IN_SECONDS );
```

## Tested Against `master` branch?

- [x] I have verified the issue still exists in the `master` branch of VIPCS.

**Comment by GaryJones** [comment]
> global memcached cache group names

Have you got a list of those @rebeccahum?

Can you give more examples of what should be flagged, and some use cases that shouldn't?

**Comment by rebeccahum** [comment]
In [the readme.txt](https://github.com/Automattic/wp-memcached/blob/8dcd4eed746cda736f91dcfe94f61ea8e9ed2a5e/readme.txt), the list seems to be here:

```
{$taxonomy}_relationships
{$meta_type}_meta
{$taxonomy}_relationships
blog-details
blog-id-cache
blog-lookup
bookmark
calendar
category
comment
counts
general
global-posts
options
plugins
post_ancestors
post_meta
posts
rss
site-lookup
site-options
site-transient
terms
themes
timeinfo
transient
user_meta
useremail
userlogins
usermeta
users
userslugs
widget
```

**Comment by GaryJones** [comment]
So, new sniff, look for `wp_cache_set` function calls, check third arg is one of those from the list, and give an Error, severity 5.

**Comment by GaryJones** [comment]
Noting that @WPprodigy is working on https://github.com/automattic/wp-cache-memcached as a replacement / update to https://github.com/automattic/wp-memcached so the global group names may change.

**Comment by WPprodigy** [comment]
Hmm, I'm not entirely sure we want to prevent this. There could be valid use cases for using one of the existing global groups in a safe way.

**Comment by GaryJones** [comment]
That's fine with me. Thanks!
